### PR TITLE
Add details on text facet filtering

### PR DESCRIPTION
**Pull request by amanifarooque**
This update reflect the text-field filtering we have added to our results API.

Preview URL: https://federalist-ecc58765-2903-48b3-920c-5d93318ad084.sites.pages.cloud.gov/preview/gsa/open-gsa-redesign/afarooque/search-gov-facets-updates/api/searchgov-results/

### CODE REVIEW

File: _apidocs/searchgov-results.md [hunk 1]
```diff
@@ -66,12 +66,30 @@ Sites indexed via sitemaps or crawling will use the `/i14y` endpoint. Because mo
...
 
+  To enable faceted search, use the parameters below. You must set `include_facets=true` for these parameters to be applied. For more context on how we index content into these fields, see our [metadata documentation](https://search.gov/indexing/metadata.html).
+
+  Facet functionality is only available to the **/i14y** endpoint. 
+
+  | Parameters                      | Description
+  | :--								| :--
+  | include_facets      			| The default is `false`. To enable facet features, set this to `true` 
+  | audience                        | Returns results that match any of the specified terms in the Audience field. Ex. `audience=students, policymakers` would return results with an audience of students **or** policymakers. Accepts a comma-separate list of strings. 
+  | content_type                    | Returns results that match any of the specified terms in the Content Type field. Ex. `mime_type=article, blog post` would return results with a content type of article **or** blog post. Accepts a comma-separate list of strings.
```
Comment (dawnpm) [inline _apidocs/searchgov-results.md:77]: this example says mime_type

File: _apidocs/searchgov-results.md [hunk 2]
```diff
@@ -66,12 +66,30 @@ Sites indexed via sitemaps or crawling will use the `/i14y` endpoint. Because mo
...
+
+  | Parameters                      | Description
+  | :--								| :--
+  | include_facets      			| The default is `false`. To enable facet features, set this to `true` 
+  | audience                        | Returns results that match any of the specified terms in the Audience field. Ex. `audience=students, policymakers` would return results with an audience of students **or** policymakers. Accepts a comma-separate list of strings. 
+  | content_type                    | Returns results that match any of the specified terms in the Content Type field. Ex. `mime_type=article, blog post` would return results with a content type of article **or** blog post. Accepts a comma-separate list of strings. 
+  | mime_type                       | Returns results that match any of the specified terms in the mime_type field, which indicates the file type of the document. Ex. `content_type=text/html, application/pdf` would return results with a mimetype of text/html **or** application/pdf. Accepts a comma-separate list of strings. [Read about common mimetypes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types).
```
Comment (dawnpm) [inline _apidocs/searchgov-results.md:78]: this example says content_type

File: _apidocs/searchgov-results.md [hunk 3]
```diff
@@ -66,12 +66,30 @@ Sites indexed via sitemaps or crawling will use the `/i14y` endpoint. Because mo
...
   | sitelimit (optional)             | Limits the results to the subdomains and/or subfolders provided. Multiple values can be passed using a space-separated list. Example: `sitelimit=search.gov/about search.gov/get-started`. The values passed in the sitelimit must be in scope of the Domains list in the Search.gov Admin Center for your search site.
 
+  To enable faceted search, use the parameters below. You must set `include_facets=true` for these parameters to be applied. For more context on how we index content into these fields, see our [metadata documentation](https://search.gov/indexing/metadata.html).
+
+  Facet functionality is only available to the **/i14y** endpoint. 
+
+  | Parameters                      | Description
+  | :--								| :--
+  | include_facets      			| The default is `false`. To enable facet features, set this to `true` 
+  | audience                        | Returns results that match any of the specified terms in the Audience field. Ex. `audience=students, policymakers` would return results with an audience of students **or** policymakers. Accepts a comma-separate list of strings.
```
Comment (dawnpm) [inline _apidocs/searchgov-results.md:76]: throughout, comma-separate or comma-separated?

File: _apidocs/searchgov-results.md [hunk 4]
```diff
@@ -66,12 +66,30 @@ Sites indexed via sitemaps or crawling will use the `/i14y` endpoint. Because mo
   | sort\_by (optional)             | Allowed variables are date and relevance. The default sort is relevance. Add `sort_by=date` to sort by date.
   | sitelimit (optional)             | Limits the results to the subdomains and/or subfolders provided. Multiple values can be passed using a space-separated list. Example: `sitelimit=search.gov/about search.gov/get-started`. The values passed in the sitelimit must be in scope of the Domains list in the Search.gov Admin Center for your search site.
 
+  To enable faceted search, use the parameters below. You must set `include_facets=true` for these parameters to be applied. For more context on how we index content into these fields, see our [metadata documentation](https://search.gov/indexing/metadata.html).
+
+  Facet functionality is only available to the **/i14y** endpoint.
```
Comment (dawnpm) [inline _apidocs/searchgov-results.md:71]: available to? available at? from? using? depends on the directionality of "available" - we make it available only to the /i14y endpoint, people need to use the /i14y endpoint to offer filters to their searchers.

I suggest we change this to "Faceted search is available only through the **/i14y** endpoint."

File: _apidocs/searchgov-results.md [hunk 5]
```diff
@@ -66,12 +66,30 @@ Sites indexed via sitemaps or crawling will use the `/i14y` endpoint. Because mo
   | sort\_by (optional)             | Allowed variables are date and relevance. The default sort is relevance. Add `sort_by=date` to sort by date.
   | sitelimit (optional)             | Limits the results to the subdomains and/or subfolders provided. Multiple values can be passed using a space-separated list. Example: `sitelimit=search.gov/about search.gov/get-started`. The values passed in the sitelimit must be in scope of the Domains list in the Search.gov Admin Center for your search site.
 
+  To enable faceted search, use the parameters below. You must set `include_facets=true` for these parameters to be applied. For more context on how we index content into these fields, see our [metadata documentation](https://search.gov/indexing/metadata.html).
```
Comment (dawnpm) [inline _apidocs/searchgov-results.md:69]: I suggest we add a sentence at the beginning, "Faceted search offers a set of filters people can use to narrow their results. We support filters for tags, audience, content type, file (mime) type, and three custom use fields. To enable faceted search on your results page, ..."

File: _apidocs/searchgov-results.md [hunk 6]
```diff
@@ -99,6 +121,16 @@ Sites indexed via sitemaps or crawling will use the `/i14y` endpoint. Because mo
       | snippet          | Summary of the document
       | publication_date | Publication date of the document (not available on commercial results)
 
+      **If `include_facets=true` in your request**, the following fields may show as well. We only display fields with content, so some web results may not have all fields indicated here.
```
Comment (dawnpm) [inline _apidocs/searchgov-results.md:124]: This reads oddly to me, maybe "If you set `includes_facets=true` in your request"

"some web results" seems like individual results to me, how about "some results sets"?

File: _apidocs/searchgov-results.md [hunk 7]
```diff
@@ -99,6 +121,16 @@ Sites indexed via sitemaps or crawling will use the `/i14y` endpoint. Because mo
...
+
+      | Values       | Description
+      | :--          | :--
```
Comment (dawnpm) [inline _apidocs/searchgov-results.md:127]: audience?

File: _apidocs/searchgov-results.md [hunk 8]
```diff
@@ -99,6 +121,16 @@ Sites indexed via sitemaps or crawling will use the `/i14y` endpoint. Because mo
...
+      | content_type | Content type of the document
+      | mime_type    | Mimetype (i.e. file type) of the document
+      | tags         | Tags associated with the document
+      | searchgov_custom1, searchgov_custom2, searchgov_custom3 | Search.gov custom field content associated with the document
```
Comment (dawnpm) [inline _apidocs/searchgov-results.md:131]: Should we separate these into different rows?

**Comment by dawnpm** [comment]
Caught a couple of additional needed changes through the preview link - couldn't add them inline because those lines weren't shown in the diff.

- should we add "Search filters (facets)" to the list of modules returned in the API blob?
- update https://search.gov/manual/site-launch-guide.html to https://search.gov/get-started/site-launch-guide.html
- update https://search.gov/blog/sitemaps.html to https://search.gov/indexing/sitemaps.html
- We need to clarify the RSS feed indexing, because "Sites indexed via RSS will use the top level / endpoint, please modify your calls accordingly." is true for ODIE sites, but not true for RSS-Garfield sites. I should have caught that in the last update.

**Review by dawnpm — CHANGES_REQUESTED** [review]
Commends throughout, plus some in the conversation thread for bits that aren't showing in this diff. thanks!!

File: _apidocs/searchgov-results/v2/openapi.yml [hunk 9]
```diff
@@ -70,6 +70,48 @@ paths:
...
+            type: string
+            format: string
+        - name: content_type
+          in: query
+          description: 'Allows you to filter results based on the Content Type field. Accepts a comma-separated list of strings.'
+          required: false
+          schema:
+            type: string
+            format: string
+        - name: tags
```
Comment (dawnpm) [inline _apidocs/searchgov-results/v2/openapi.yml:87]: `tags` is below as well, `mime_type` is missing though.

Comment (dawnpm) [reply _apidocs/searchgov-results/v2/openapi.yml:87]: guessing this should be `mime_type`?

File: _apidocs/searchgov-results.md (hunk 6 above)
Comment (amanifarooque) [reply _apidocs/searchgov-results.md:124]: Hm, it is individual results though - if one result has `content_type` indicated but another doesn't, only the one with a non-null content type would have that field returned in the result object.

File: _apidocs/searchgov-results.md (hunk 3 above)
Comment (amanifarooque) [reply _apidocs/searchgov-results.md]: fixed, my bad!

File: _apidocs/searchgov-results.md (hunk 6 above)
Comment (amanifarooque) [reply _apidocs/searchgov-results.md:124]: updated with your language

File: _apidocs/searchgov-results.md (hunk 8 above)
Comment (amanifarooque) [reply _apidocs/searchgov-results.md:131]: updated

File: _apidocs/searchgov-results.md (hunk 2 above)
Comment (amanifarooque) [reply _apidocs/searchgov-results.md]: updated

File: _apidocs/searchgov-results.md (hunk 4 above)
Comment (amanifarooque) [reply _apidocs/searchgov-results.md:71]: updated

File: _apidocs/searchgov-results.md (hunk 5 above)
Comment (amanifarooque) [reply _apidocs/searchgov-results.md:69]: updated

File: _apidocs/searchgov-results.md (hunk 7 above)
Comment (amanifarooque) [reply _apidocs/searchgov-results.md:127]: updated

File: _apidocs/searchgov-results/v2/openapi.yml (hunk 9 above)
Comment (amanifarooque) [reply _apidocs/searchgov-results/v2/openapi.yml:87]: the tags below is different, it's not nested within the parameters object.
added mime_type

**Comment by amanifarooque** [comment]
Thanks Dawn for your review! Changes have been made accordingly. On your question below:

> should we add "Search filters (facets)" to the list of modules returned in the API blob?

The facet information is all housed within the existing "web" module, so this won't be necessary.

**Review by dawnpm — APPROVED** [review]

**Comment by dawnpm** [comment]
@MorganRedden would you please merge this PR? thank you.

**Pull request merged**